src/vstest.console/Processors/RunSpecificTestsArgumentProcessor.cs

```diff
@@ -218,6 +219,14 @@ public ArgumentProcessorResult Execute()
             // Now that tests are discovered and filtered, we run only those selected tests.
             this.ExecuteSelectedTests();
 
+            bool failWhenNoTestsFound = RunSettingsUtilities.GetFailWhenNoTestsFound(effectiveRunSettings);
```

Comment:
If we don't use filtering then RunTestsArgumentProcessor.cs part will execute the operation. When we run on non-test assembly it will behave the same as with test assembly without any tests.

Reply:
ok, then that should be correct.